Mathlib/MeasureTheory/Function/L1Space.lean

```diff
@@ -435,11 +436,12 @@ def Integrable {α} {_ : MeasurableSpace α} (f : α → β) (μ : Measure α :=
   AEStronglyMeasurable f μ ∧ HasFiniteIntegral f μ
 #align measure_theory.integrable MeasureTheory.Integrable
 
-theorem integrable_eq {α} {_ : MeasurableSpace α} (f : α → β) (μ : Measure α) :
-    Integrable f μ = (AEStronglyMeasurable f μ ∧ HasFiniteIntegral f μ) :=
-  rfl
+-- Porting note: Lean can't generate an equation lemma for this.
```

Comment:
I don't really understand this comment; what happens without this theorem?

Reply:
Without this theorem, `rw [Integrable]` doesn't work correctly.
```lean
theorem integrable_const_iff {c : β} : Integrable (fun _ : α => c) μ ↔ c = 0 ∨ μ univ < ∞ := by
  have : AEStronglyMeasurable (fun _ : α => c) μ := aestronglyMeasurable_const
  rw [Integrable, and_iff_right this, hasFiniteIntegral_const_iff]
-- failed to rewrite using equation theorems for 'MeasureTheory.Integrable'
```

Reply:
Can you check what the equation lemmas it does generate are? I think we should be clear in the comment whether the problem is:
* The equation lemma generated is nonsense
* No equation lemmas are generated

Reply:
An equation lemma is generated correctly, but when it is used in `rw`, the equation lemma is re-elaborated so the `μ` argument is filled by `volume_tac`, and the error occurs because there are no `MeasureSpace` instance.
I will describe this in Porting note.

Reply:
I think this is worth opening an issue (or at least, a zulip thread) about. There's no way that this behavior is useful.

Reply:
I've submitted the [issue](https://github.com/leanprover/lean4/issues/2243).

Reply:
My explanation was wrong. Indeed, `rewrite` fails even without `MeasureSpace` dependency:
```lean
import Lean

open Lean Elab Tactic in
elab "exact_false" : tactic =>
  closeMainGoal (mkConst ``Bool.false)

def f (b : Bool := by exact_false) : Nat := bif b then 1 else 0

example : f false = bif false then 1 else 0 := by rw [f]
example : f true = bif true then 1 else 0 := by rw [f]
example (b : Bool) : f b = bif b then 1 else 0 := by rw [f]
```